app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -210,14 +216,129 @@ class SmartTransactionHook {
     }
   }
 
+  async submitBatch() {
+    // Will cause TransactionController to publish to the RPC provider as normal.
+    const useRegularTransactionSubmit = undefined;
+
+    if (!this.#isSmartTransaction) {
+      return useRegularTransactionSubmit;
+    }
+
+    if (this.#shouldShowStatusPage) {
+      await this.#startApprovalFlow();
+    }
+
+    try {
+      const submitTransactionResponse = await this.#signAndSubmitTransactions();
+      const uuid = submitTransactionResponse?.uuid;
+
+      if (!uuid) {
+        throw new Error('No smart transaction UUID');
+      }
+
+      await this.#processApprovalIfNeeded(uuid);
+
+      const transactionHash = await this.#waitForTransactionHash({
+        uuid,
+      });
+
+      if (transactionHash === null) {
+        throw new Error(
+          'Transaction does not have a transaction hash, there was a problem',
+        );
+      }
+
+      let submitBatchResponse;
+      if (submitTransactionResponse?.txHashes) {
```

Comment:
Minor, ternary instead of `let`?

Reply:
I've tried that after your suggestion, but it seemed little less readable. Also, AI mentioned:
_While the ternary operator is more concise, when dealing with complex nested operations like the map function inside object literals, the traditional if/else structure makes the conditional logic clearer and easier to scan._